lib/cli/src/commands/publish.rs

```diff
@@ -0,0 +1,652 @@
+            Err(e) => {
+                log::error!(
+                    "Could not read private key from location {}: {}",
+                    priv_key_location,
+                    e
+                );
+                return Err(e.into());
+            }
+        }
+    } else {
+        // TODO: add more info about why this might have happened and what the user can do about it
+        log::warn!("Active key does not have a private key location registered with it!");
+        return Err(anyhow!("Cannot sign package, no private key"));
+    };
+    Ok(SignArchiveResult::Ok {
+        public_key_id: personal_key.public_key_id,
+        signature: (minisign::sign(
+            Some(&minisign::PublicKey::from_base64(
+                &personal_key.public_key_value,
+            )?),
+            &private_key,
+            compressed_archive,
+            None,
+            None,
+        )?
+        .to_string()),
+    })
+}
+
+/// Opens an exclusive read/write connection to the database, creating it if it does not exist
+pub fn open_db() -> anyhow::Result<Connection> {
+    let db_path =
+        PartialWapmConfig::get_database_file_path().map_err(|e| anyhow::anyhow!("{e}"))?;
+    let mut conn = Connection::open_with_flags(
+        db_path,
+        OpenFlags::SQLITE_OPEN_CREATE
+            | OpenFlags::SQLITE_OPEN_READ_WRITE
+            | OpenFlags::SQLITE_OPEN_FULL_MUTEX,
+    )?;
+
+    apply_migrations(&mut conn)?;
+    Ok(conn)
+}
+
+/// Applies migrations to the database
+pub fn apply_migrations(conn: &mut Connection) -> anyhow::Result<()> {
+    let user_version = conn.pragma_query_value(None, "user_version", |val| val.get(0))?;
+    for data_version in user_version..CURRENT_DATA_VERSION {
+        log::debug!("Applying migration {}", data_version);
+        apply_migration(conn, data_version)?;
+    }
+    Ok(())
+}
+
+#[derive(Debug, Error)]
+enum MigrationError {
+    #[error(
+        "Critical internal error: the data version {0} is not handleded; current data version: {1}"
+    )]
+    MigrationNumberDoesNotExist(i32, i32),
+    #[error("Critical internal error: failed to commit trasaction migrating to data version {0}")]
+    CommitFailed(i32),
+    #[error("Critical internal error: transaction failed on migration number {0}: {1}")]
+    TransactionFailed(i32, String),
+}
+
+/// Applies migrations to the database and updates the `user_version` pragma.
+/// Every migration must leave the database in a valid state.
+fn apply_migration(conn: &mut Connection, migration_number: i32) -> Result<(), MigrationError> {
+    let tx = conn
+        .transaction_with_behavior(TransactionBehavior::Immediate)
+        .map_err(|e| MigrationError::TransactionFailed(migration_number, format!("{}", e)))?;
+
+    let migrations = &[
+        (0, include_str!("../../sql/migrations/0000.sql")),
+        (1, include_str!("../../sql/migrations/0001.sql")),
+        (2, include_str!("../../sql/migrations/0002.sql")),
+    ];
```

Comment:
If we hoist this into its own `const` variable at the top of the file, `CURRENT_DATA_VERSION` just becomes `MIGRATIONS.len()`.

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3432/commits/e9ceff9e1620aa33c0f1327c0ddfb9af4440b3e5